tests/ducktape/test_consumer.py

```
@@ -147,73 +185,245 @@ def test_basic_consume(self, consumer_type, batch_size):
+            assignment1_final = await consumer1.assignment()
+            assignment2_final = await consumer2.assignment()
+            assignment3_final = await consumer3.assignment()
+            total_partitions_final = (len(assignment1_final) +
+                                      len(assignment2_final) +
+                                      len(assignment3_final))
+
+            assert total_partitions_final == 4, \
+                f"Should have 4 total partitions after adding, got {total_partitions_final}"
+            # With 3 consumers and 4 partitions, distribution should be roughly 1-2 partitions per consumer
+            assert len(assignment1_final) >= 1, \
+                f"Consumer 1 should have at least 1 partition, got {len(assignment1_final)}"
+            assert len(assignment2_final) >= 1, \
+                f"Consumer 2 should have at least 1 partition, got {len(assignment2_final)}"
+            assert len(assignment3_final) >= 1, \
+                f"Consumer 3 should have at least 1 partition, got {len(assignment3_final)}"
+            assert len(rebalance_events) >= 5, \
+                ("Should have at least 5 rebalance events after partition addition and consumer3 join, "
+                 f"got {len(rebalance_events)}")
+
+            # Verify consumers can still consume from all partitions
+            msg1 = await consumer1.poll(timeout=5.0)
+            msg2 = await consumer2.poll(timeout=5.0)
+            msg3 = await consumer3.poll(timeout=5.0)
+            messages_received = sum([1 for msg in [msg1, msg2, msg3] if msg is not None])
+            assert messages_received > 0, "Consumers should receive messages from new partitions"
+
+            # Clean up
+            await consumer1.close()
+            await consumer2.close()
+            await consumer3.close()
+
+        asyncio.run(async_topic_change_test())
+
+    # TODO: verify if the current behavior is correct/intended
```

I would be surprised if it swallowed a callback error without reraising tbh